### DISCUSSION TITLE
Table 5 Panel 3

### DISCUSSION THREAD

**Discussion by scleveland** [cssi-pi-community/2022-meeting#89] [Panel 3 roundtable (Tuesday morning)]
Coding in haste and not enough coding documentation.  Missing workflow/introduction increases the overhead for the next maintainer or user to adopt/develop.

- Creating a doc standard for scientific codes
- More training programs (like carpentries.org efforts) for basic background for researchers that need to code
- Fundamental programming/course at Univesities where students can learn standards/best-practices. Get student/researchers into these mindsets early.
-

**Comment by scleveland** [comment]
Resources to help faculty/researchers scale their work.  It is difficult to hire/find professional level developers/computational analysts that can do less than full FTE or for 1 or 2 years on soft-funding and uncertainty.
- NSF supplied coders or center that researchers can access
- University resources and centers with developers that can offer consulting services in support of research with staff that understand research or convert algorithm to code, packaging code/licensing etc

**Comment by scleveland** [comment]
Being able to generalize models from specific use cases so other can leverage and actually being able to find these models/algorithms that exist.
- Some discovery tool or AI/helper to query and get suggestions for matches with labs/software/resources would help 
- Some incentives/resources for the original researcher/developer to migrate/implement for broader adoption of applicable codes

**Comment by echoi** [comment]
> Coding in haste and not enough coding documentation. Missing workflow/introduction increases the overhead for the next maintainer or user to adopt/develop.
> 
> * Creating a doc standard for scientific codes
> * More training programs (like carpentries.org efforts) for basic background for researchers that need to code
> * Fundamental programming/course at Univesities where students can learn standards/best-practices. Get student/researchers into these mindsets early.

Services like Software Carpentry (https://carpentries.org/) can help but not directly provide training on documentation.
Is there a resource or a standard for systematic software documentation not only API documentation and in-code comments but also a user manual?
